torch/csrc/autograd/python_variable.cpp

```diff
@@ -35,11 +41,13 @@ static PyObject* THPVariable_NewWithVar(PyTypeObject* type, Variable var)
   if (obj) {
     auto v = (THPVariable*) obj;
     new (&v->cdata) Variable(std::move(var));
-    v->cdata.get()->pyobj = obj;
-    if (auto fn = dynamic_cast<PyFunction*>(v->cdata.get()->_grad_fn.get())) {
+    v->cdata.set_pyobj(obj);
+    if (auto fn = std::dynamic_pointer_cast<PyFunction>(v->cdata.grad_fn())) {
```

Comment:
This previously didn't require a temporary incref, but now it does

Reply:
I think this also will modify the `_grad_fn` on stale views, which I don't think we want to do here.